AnkiDroid/src/main/java/com/ichi2/anki/Sync.kt

```diff
@@ -72,7 +73,12 @@ fun DeckPicker.handleNewSync(
             when (conflict) {
                 Connection.ConflictResolution.FULL_DOWNLOAD -> handleDownload(deckPicker, auth, syncMedia)
                 Connection.ConflictResolution.FULL_UPLOAD -> handleUpload(deckPicker, auth, syncMedia)
-                null -> handleNormalSync(deckPicker, auth, syncMedia)
+                null -> {
+                    try { handleNormalSync(deckPicker, auth, syncMedia) } catch (exc: Exception) {
+                        showSnackbar(R.string.hostname_exception)
+                        Timber.i("No network exception")
```

Comment:
Out of curiosity on this, why catch all Exception instead of just the recommended exceptions from @BrayanDSO ?

https://github.com/ankidroid/Anki-Android/issues/12711#issuecomment-1296352330

> My suggestion is catching UnknownHostException and BackendNetworkException exceptions and dealing with them

Catching Exception is generally discouraged when you have specific exceptions you can catch, especially if you simply "eat" the exception, meaning the user will maybe have some unexpected exception mapped to a "network" error snackbar when it could be something more serious, and we will never get the crash trace for it

https://stackoverflow.com/a/36760590/9910298

```suggestion
                        when (exc) {
                            is UnknownHostException, is BackendNetworkException -> {
                                showSnackbar(R.string.hostname_exception)
                                Timber.i("No network exception")
                            }
                            else -> throw exc
```

...I think it goes something like that to catch multiple exception types but ignore rest in Kotlin?

Reply:
true, more relevant will implement that.